src/sensors/distant.cpp

```diff
@@ -0,0 +1,494 @@
+template <typename Float, typename Spectrum, RayTargetType TargetType,
+          RayOriginType OriginType>
+class DistantSensorImpl;
+
+/**!
+
+.. _sensor-distant:
+
+Distant directional sensor (:monosp:`distant`)
+----------------------------------------------
+
+.. pluginparameters::
+
+ * - to_world
+   - |transform|
+   - Sensor-to-world transformation matrix.
+ * - direction
+   - |vector|
+   - Alternative (and exclusive) to ``to_world``. Direction orienting the sensor's
+     reference hemisphere.
+ * - orientation
+   - |vector|
+   - If ``direction`` is set, this vector parameter can be used to control
+     film orientation. The orientation is given by the projection of
+     ``orientation`` onto the  plane of normal ``direction``, and the up vector
+     defined as :math:`\mathrm{up} = \mathrm{direction} \times \mathrm{orientation}`.
+     If ``direction`` is unset, this parameter has no effect.
+ * - flip_directions
+   - |bool|
+   - If true, flip the directions of sampled rays. Default: false
+ * - ray_target
+   - |point| or nested :paramtype:`shape` plugin
+   - *Optional.* Define the ray target sampling strategy.
+     If this parameter is unset, ray target points are sampled uniformly on
+     the cross section of the scene's bounding sphere.
+     If a |point| is passed, rays will target it.
+     If a shape plugin is passed, ray target points will be sampled from its
+     surface.
+ * - ray_origin
+   - nested :paramtype:`shape` plugin
+   - *Optional.* Specify the ray origin computation strategy.
+     If this parameter is unset, ray origins will be positioned using the
+     bounding sphere of the scene so as to ensure that they lie outside of any
+     geometry.
+     If a shape plugin is passed, ray origin points will be positioned by
+     projecting the sampled target point onto the shape following the sampled
+     ray direction. If the projection is impossible, an invalid ray is returned
+     with zero weights. *Note*: ray invalidation occurs per-lane in packet
+     modes.
+
+This sensor plugin implements a distant directional sensor which records
+radiation leaving the scene in a given direction. It records the spectral
+radiance leaving the scene in the specified direction.
+Rays target points are sampled from the cross section of the scene's bounding
+sphere and their origins are positioned outside of the scene's geometry.
+
+Based on the film size, the ray direction sampling strategy will vary:
+
+* if a 1x1 film is passed, rays directions will be equal to ``-direction``
+  (unless ``flip_directions`` is true, in which case they will be equal to
+  ``direction``);
+* if a Nx1 film is passed (*i.e.* if film height is reduced to 1), ray
+  directions will be sampled from the intersection of the hemisphere defined by
+  ``-direction`` and the (vector) plane generated by ``orientation`` and
+  ``direction``;
+* if a NxM film is passed, ray directions will be sampled in the hemisphere
+  defined by ``-direction``.
```

Comment:
If I understand correctly, this sensor would only really be *directional* when the film as a size of `1x1`?

In which case I would say that we might remove the "directional" word in the name of this plugin.

Reply:
Good point!